service/worker/deployment/deployment_activities.go

```diff
@@ -116,3 +117,33 @@ func (a *DeploymentActivities) CheckUserDataPropagation(ctx context.Context, inp
 	}
 	return err
 }
+
+func (a *DeploymentActivities) IsDeploymentDrainedOfOpenWorkflows(ctx context.Context, input *deploymentspb.IsDeploymentDrainedOfOpenWorkflowsRequest) (*deploymentspb.IsDeploymentDrainedOfOpenWorkflowsResponse, error) {
+	logger := activity.GetLogger(ctx)
+
+	response, err := a.deploymentClient.IsDeploymentDrainedOfOpenWorkflows(ctx, a.namespace, input.Deployment)
+	if err != nil {
+		logger.Error("error when checking if deployment reachability", "error", err)
+		return nil, err
+	}
+	return &deploymentspb.IsDeploymentDrainedOfOpenWorkflowsResponse{IsDrained: response}, nil
+}
+
+func (a *DeploymentActivities) IsDeploymentDrainedOfAllWorkflows(ctx context.Context, input *deploymentspb.IsDeploymentDrainedOfAllWorkflowsRequest) (*deploymentspb.IsDeploymentDrainedOfAllWorkflowsResponse, error) {
+	logger := activity.GetLogger(ctx)
+
+	response, err := a.deploymentClient.IsDeploymentDrainedOfAllWorkflows(ctx, a.namespace, input.Deployment)
+	if err != nil {
+		logger.Error("error when checking if deployment is scavenged", "error", err)
+		return nil, err
+	}
+	return &deploymentspb.IsDeploymentDrainedOfAllWorkflowsResponse{IsDrained: response}, nil
+}
+
+func (a *DeploymentActivities) GetNamespaceRetention() (time.Duration, error) {
```

Comment:
better to use a [SideEffect](https://docs.temporal.io/develop/java/side-effects#side-effects) for this.